resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala

```diff
@@ -143,17 +143,29 @@ private object ResourceRequestHelper extends Logging {
     require(resource != null, "Resource parameter should not be null!")
 
     logDebug(s"Custom resources requested: $resources")
+    if (resources.isEmpty) {
+      // no point in going forward, as we don't have anything to set
+      return
+    }
+
     if (!isYarnResourceTypesAvailable()) {
-      if (resources.nonEmpty) {
-        logWarning("Ignoring custom resource requests because " +
-            "the version of YARN does not support it!")
-      }
+      logWarning("Ignoring custom resource requests because " +
+          "the version of YARN does not support it!")
       return
     }
 
     val resInfoClass = Utils.classForName(RESOURCE_INFO_CLASS)
     val setResourceInformationMethod =
-      resource.getClass.getMethod("setResourceInformation", classOf[String], resInfoClass)
+      try {
+        resource.getClass.getMethod("setResourceInformation", classOf[String], resInfoClass)
+      } catch {
+        case e: NoSuchMethodException =>
+          throw new SparkException(
+            s"""Cannot find $RESOURCE_INFO_CLASS.setResourceInformation.
+                |This is likely due to a jar conflict between different yarn versions."""
+                .stripMargin.replace("\n", " "), e)
```

Comment:
nit. The following will be simpler instead of having `stripMargin.replace("\n", " ")`.
```scala
throw new org.apache.spark.SparkException(
  s"Cannot find $RESOURCE_INFO_CLASS.setResourceInformation. " +
    "This is likely due to a jar conflict between different yarn versions.", e)
```

Reply:
@dongjoon-hyun I changed this, sorry for the delay.